tools/windows/DatadogAgentInstaller/CustomActions/InstallStateCustomActions.cs

```diff
@@ -154,6 +160,61 @@ public void GetWindowsBuildVersion()
             }
         }
 
+        public void SetDDDriverRollback()
+        {
+            var upgradeDetected = _session["WIX_UPGRADE_DETECTED"];
+
+            if (!string.IsNullOrEmpty(upgradeDetected))
+            {
+                var versionString = _nativeMethods.GetVersionString(upgradeDetected);
+                // Using Version class 
+                // https://learn.microsoft.com/en-us/dotnet/api/system.version?view=net-8.0
+                var currentVersion = new Version(versionString);
+                Version version_53;
+                Version minimumVersion;
+
+                // Check major version
+                if (versionString[0] == '7')
+                {
+                    version_53 = new Version("7.53");
+                    minimumVersion = new Version("7.56");
+                }
+                else
+                {
+                    version_53 = new Version("6.53");
+                    minimumVersion = new Version("6.56");
+                }
+
+                var compareResult = currentVersion.CompareTo(minimumVersion);
+                if (compareResult < 0) // currentVersion is less than minimumVersion
+                {
```

Comment:
Use a more descriptive comment
```suggestion
                if (compareResult < 0)
                {
                    // upgrading from a version that didn't implement driver rollback
                    // Clear NPM flag to ensure NPM service is not deleted on rollback
```